### ISSUE TITLE
Unprivileged container cannot set lxc.cgroup2.something under host with systemd.unified_cgroup_hierarchy

### ISSUE THREAD

**Issue by emojifreak** [lxc/lxc#3201]
The below is observed on Ubuntu 19.10 with systemd.unified_cgroup_hierarchy, lxc version
```
liblxc-common	3.2.1+master~20191202-0205-0ubuntu1~eoan
liblxc1	3.2.1+master~20191202-0205-0ubuntu1~eoan
lxc-templates	3.0.4-1ubuntu1
lxc-utils	3.2.1+master~20191202-0205-0ubuntu1~eoan
```

In a privileged container on host Linux with systemd.unified_cgroup_hierarchy, I verified that 

```
lxc.cgroup2.memory.high = 2G
lxc.cgroup2.pids.max = 5000
```

correctly set the resource control, when lxc.cgroup.relative is **unset**. The resource control fails with `lxc.cgroup.relative=1`, but let us ignore it.

Since `libpam-cgfs` seems unworking under systemd.unified_cgroup_hierarchy (see https://github.com/lxc/lxc/issues/3198), I started an unprivileged container with `systemd-run --unit=mytest --user --scope lxc-start -F -n unpriv-eoan --logpriority=INFO'. The config file has

```
lxc.cgroup.devices.deny =
lxc.cgroup.devices.allow =
lxc.cgroup2.devices.deny =
lxc.cgroup2.devices.allow =
lxc.mount.auto = cgroup:rw:force
lxc.cgroup2.memory.high = 2G
lxc.cgroup2.pids.max = 5000
```

Then the unprivileged container does not start with the error log

```
lxc-start unpriv-eoan 20191203125822.422 INFO     confile - confile.c:set_config_idmaps:2003 - Read uid map: type u nsid 0 hostid 100000 range 65536
lxc-start unpriv-eoan 20191203125822.422 INFO     confile - confile.c:set_config_idmaps:2003 - Read uid map: type g nsid 0 hostid 100000 range 65536
lxc-start unpriv-eoan 20191203125822.422 INFO     lsm - lsm/lsm.c:lsm_init:50 - LSM security driver AppArmor
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "reject_force_umount  # comment this to allow umount -f;  not recommended"
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:do_resolve_add_rule:535 - Set seccomp rule to reject force umounts
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:973 - Added native rule for arch 0 for reject_force_umount action 0(kill)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:do_resolve_add_rule:535 - Set seccomp rule to reject force umounts
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:982 - Added compat rule for arch 1073741827 for reject_force_umount action 0(kill)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:do_resolve_add_rule:535 - Set seccomp rule to reject force umounts
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:992 - Added compat rule for arch 1073741886 for reject_force_umount action 0(kill)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:do_resolve_add_rule:535 - Set seccomp rule to reject force umounts
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:1002 - Added native rule for arch -1073741762 for reject_force_umount action 0(kill)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "[all]"
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "kexec_load errno 1"
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:973 - Added native rule for arch 0 for kexec_load action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:982 - Added compat rule for arch 1073741827 for kexec_load action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:992 - Added compat rule for arch 1073741886 for kexec_load action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:1002 - Added native rule for arch -1073741762 for kexec_load action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "open_by_handle_at errno 1"
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:973 - Added native rule for arch 0 for open_by_handle_at action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:982 - Added compat rule for arch 1073741827 for open_by_handle_at action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:992 - Added compat rule for arch 1073741886 for open_by_handle_at action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:1002 - Added native rule for arch -1073741762 for open_by_handle_at action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "init_module errno 1"
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:973 - Added native rule for arch 0 for init_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:982 - Added compat rule for arch 1073741827 for init_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:992 - Added compat rule for arch 1073741886 for init_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:1002 - Added native rule for arch -1073741762 for init_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "finit_module errno 1"
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:973 - Added native rule for arch 0 for finit_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:982 - Added compat rule for arch 1073741827 for finit_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.422 INFO     seccomp - seccomp.c:parse_config_v2:992 - Added compat rule for arch 1073741886 for finit_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:1002 - Added native rule for arch -1073741762 for finit_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:789 - Processing "delete_module errno 1"
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:973 - Added native rule for arch 0 for delete_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:982 - Added compat rule for arch 1073741827 for delete_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:992 - Added compat rule for arch 1073741886 for delete_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:1002 - Added native rule for arch -1073741762 for delete_module action 327681(errno)
lxc-start unpriv-eoan 20191203125822.423 INFO     seccomp - seccomp.c:parse_config_v2:1008 - Merging compat seccomp contexts into main context
lxc-start unpriv-eoan 20191203125822.596 ERROR    cgroup2_devices - cgroups/cgroup2_devices.c:bpf_devices_cgroup_supported:514 - Invalid argument - The bpf device cgroup requires real root
lxc-start unpriv-eoan 20191203125822.596 INFO     start - start.c:lxc_init:942 - Container "unpriv-eoan" is initialized
lxc-start unpriv-eoan 20191203125822.597 INFO     cgfsng - cgroups/cgfsng.c:cgfsng_monitor_create:1423 - The monitor process uses "lxc.monitor/unpriv-eoan" as cgroup
lxc-start unpriv-eoan 20191203125822.597 INFO     cgfsng - cgroups/cgfsng.c:cgfsng_payload_create:1488 - The container process uses "lxc.payload/unpriv-eoan" as cgroup
lxc-start unpriv-eoan 20191203125822.598 INFO     start - start.c:lxc_spawn:1799 - Cloned CLONE_NEWUSER
lxc-start unpriv-eoan 20191203125822.598 INFO     start - start.c:lxc_spawn:1799 - Cloned CLONE_NEWNS
lxc-start unpriv-eoan 20191203125822.598 INFO     start - start.c:lxc_spawn:1799 - Cloned CLONE_NEWPID
lxc-start unpriv-eoan 20191203125822.598 INFO     start - start.c:lxc_spawn:1799 - Cloned CLONE_NEWUTS
lxc-start unpriv-eoan 20191203125822.598 INFO     start - start.c:lxc_spawn:1799 - Cloned CLONE_NEWIPC
lxc-start unpriv-eoan 20191203125822.608 INFO     start - start.c:do_start:1203 - Unshared CLONE_NEWNET
lxc-start unpriv-eoan 20191203125822.608 ERROR    cgfsng - cgroups/cgfsng.c:__cg_unified_setup_limits:2665 - Permission denied - Failed to set "memory.high" to "2G"
lxc-start unpriv-eoan 20191203125822.609 ERROR    start - start.c:lxc_spawn:1840 - Failed to setup cgroup limits for container "unpriv-eoan"
lxc-start unpriv-eoan 20191203130519.503 ERROR    start - start.c:__lxc_start:2069 - Failed to spawn container "unpriv-eoan"
```

**Comment by emojifreak** [comment]
I observed that an unprivileged container started by `root` without `lxc.cgroup.relative` can set CGroup V2 resource control such as lxc.cgroup2.memory.high and lxc.cgroup2.pids.max. But if `lxc.cgroup.relative=1` is set, the container fails to start with the exactly the same error log above.

**Comment by brauner** [comment]
I'm guessing that the `pids` and `memory` controller are not enabled in your current cgroup where you call `lxc-start` from. You can look at the controllers you have available by looking at the `cgroup.controllers` file.
`lxc` will delegate controllers if it can but if you or your system administrator have not enabled controllers for delegation in the cgroup root and you ask `lxc` to stay in the root cgroup with `lxc.cgroup.relative = 1` there's nothing that `lxc` can do.

**Comment by brauner** [comment]
That's simply how the new delegation model works.

**Comment by emojifreak** [comment]
@brauner Thank you for your comment. But
>That's simply how the new delegation model works.

I know it.

> I'm guessing that the pids and memory controller are not enabled in your current cgroup where you call lxc-start from.

**IT IS ENABLED.** Please have a look at the following transcript of commands. **I STRONGLY BELIEVE THAT LXC HAS A PROBLEM IN THIS ASPECT.**

```
ryutaroh@oldlaptop:~$ systemd-run --user --scope --unit=myshell -p "Delegate=memory pids" /bin/bash
Running scope as unit: myshell.scope
ryutaroh@oldlaptop:~$ systemctl --user status myshell.scope
● myshell.scope - /bin/bash
   Loaded: loaded (/run/user/1000/systemd/transient/myshell.scope; transient)
Transient: yes
   Active: active (running) since Wed 2019-12-04 07:30:21 JST; 15s ago
    Tasks: 2 (limit: 4541)
   Memory: 1.2M
      CPU: 10ms
   CGroup: /user.slice/user-1000.slice/user@1000.service/myshell.scope
           ├─1669 /bin/bash
           └─1681 systemctl --user status myshell.scope

ryutaroh@oldlaptop:~$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/cgroup.controllers 
io memory pids

ryutaroh@oldlaptop:~$ lxc-start -F -n unpriv-eoan --logpriority=INFO --logfile=eoan-unpriv111.log
lxc-start: unpriv-eoan: cgroups/cgroup2_devices.c: bpf_devices_cgroup_supported: 514 Invalid argument - The bpf device cgroup requires real root
lxc-start: unpriv-eoan: cgroups/cgfsng.c: __cg_unified_setup_limits: 2665 Permission denied - Failed to set "pids.max" to "5000"
lxc-start: unpriv-eoan: start.c: lxc_spawn: 1840 Failed to setup cgroup limits for container "unpriv-eoan"
lxc-start: unpriv-eoan: start.c: __lxc_start: 2069 Failed to spawn container "unpriv-eoan"
lxc-start: unpriv-eoan: tools/lxc_start.c: main: 329 The container failed to start
lxc-start: unpriv-eoan: tools/lxc_start.c: main: 334 Additional information can be obtained by setting the --logfile and --logpriority options
```
After lxc failed, lsc left some directories under `myshell.scope` cgroup as below:

```
ryutaroh@oldlaptop:~$ ls -lR /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope
/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope:
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.controllers
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.events
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.freeze
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.max.depth
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.max.descendants
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.procs
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.subtree_control
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.threads
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cgroup.type
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cpu.pressure
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 cpu.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 io.max
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 io.pressure
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 io.stat
drwxr-xr-x 2 ryutaroh ryutaroh 0 Dec  4 07:32 lxc.monitor
drwxr-xr-x 2 ryutaroh ryutaroh 0 Dec  4 07:32 lxc.payload
drwxr-xr-x 2 ryutaroh ryutaroh 0 Dec  4 07:32 lxc.pivot
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.current
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.events
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.events.local
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.high
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.low
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.max
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.min
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.oom.group
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.pressure
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 memory.stat
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 pids.current
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 pids.events
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:30 pids.max

/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor:
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.controllers
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.events
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.freeze
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.max.depth
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.max.descendants
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.procs
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.subtree_control
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.threads
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.type
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cpu.pressure
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cpu.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 io.pressure
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 memory.pressure

/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.payload:
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.controllers
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.events
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.freeze
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.max.depth
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.max.descendants
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.procs
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.subtree_control
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.threads
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.type
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cpu.pressure
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cpu.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 io.pressure
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 memory.pressure

/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.pivot:
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.controllers
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.events
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.freeze
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.max.depth
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.max.descendants
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.procs
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.subtree_control
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.threads
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cgroup.type
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cpu.pressure
-r--r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 cpu.stat
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 io.pressure
-rw-r--r-- 1 ryutaroh ryutaroh 0 Dec  4 07:32 memory.pressure
```

**Comment by brauner** [comment]
It's enabled in your cgroup's:
```
cgroup.controllers
```
file but not in your cgroup's
```
cgroup.subtree_control
```
file. And `lxc` currently will not tamper with your `cgroup.subtree_control` settings.

I might change that though...

**Comment by brauner** [comment]
And we can't really tamper with `cgroup.subtree_control` for some controllers such as `memory` when the cgroup is already populated even if it is delegated to us.

**Comment by emojifreak** [comment]
@brauner Thank you very much for your quick response.

If I read your comment correctly, it seems that enablement of cgroup.subtree_control is a job of libpam_cgfs...

But `subtree_control` did not help (see below) even when I can enable a controller by hand. It caused even strange behavior. Note that below I only try to enable `lxc.cgroup2.pids.max = 5000`.

```
ryutaroh@oldlaptop:~$ systemd-run --user --scope --unit=myshell -p "Delegate=memory pids" /bin/bash
Running scope as unit: myshell.scope
ryutaroh@oldlaptop:~$ tail ~/.local/share/lxc/unpriv-eoan/config

lxc.cgroup.devices.deny =
lxc.cgroup.devices.allow =
lxc.cgroup2.devices.deny =
lxc.cgroup2.devices.allow =
lxc.mount.auto = cgroup:rw:force
lxc.init.cmd = /bin/bash
#lxc.cgroup.relative = 1
#lxc.cgroup2.memory.high = 2G
lxc.cgroup2.pids.max = 5000
ryutaroh@oldlaptop:~$ systemctl --user status myshell.scope
● myshell.scope - /bin/bash
   Loaded: loaded (/run/user/1000/systemd/transient/myshell.scope; transient)
Transient: yes
   Active: active (running) since Wed 2019-12-04 08:19:40 JST; 46s ago
    Tasks: 2 (limit: 4541)
   Memory: 1.4M
      CPU: 14ms
   CGroup: /user.slice/user-1000.slice/user@1000.service/myshell.scope
           ├─1637 /bin/bash
           └─1654 systemctl --user status myshell.scope

Dec 04 08:19:40 oldlaptop systemd[845]: Started /bin/bash.
ryutaroh@oldlaptop:~$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/cgroup.controllers 
io memory pids
ryutaroh@oldlaptop:~$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/cgroup.subtree_control 
ryutaroh@oldlaptop:~$ echo +pids > /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/cgroup.subtree_control 
ryutaroh@oldlaptop:~$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/cgroup.subtree_control 
pids
ryutaroh@oldlaptop:~$ lxc-start -F -n unpriv-eoan --logpriority=INFO --logfile=eoan-unpriv112.log
lxc-start: unpriv-eoan: cgroups/cgroup2_devices.c: bpf_devices_cgroup_supported: 514 Invalid argument - The bpf device cgroup requires real root
lxc-start: unpriv-eoan: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv-eoan: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv-eoan"
lxc-start: unpriv-eoan: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv-eoan: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv-eoan-1"
lxc-start: unpriv-eoan: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv-eoan: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv-eoan-2"
```

The above continues 1000 times. Because the transcript of commands is too big, I make it as an attachment.

[transcript.txt](https://github.com/lxc/lxc/files/3919176/transcript.txt)
[eoan-unpriv112.log](https://github.com/lxc/lxc/files/3919177/eoan-unpriv112.log)

**Comment by emojifreak** [comment]
@brauner The indefinite loop in LXC written in my last comment occurs without `lxc.cgroup2.pids.max`. Please have a look at the below transcript. The condition is that `cgroup.subtree_control` has pids. Could you consider to **reopen** this issue?

```
ryutaroh@newlaptop:~$ export MY_CGROUP=/sys/fs/cgroup/user.slice/user-${UID}.slice/user@${UID}.service/myshell.scope

ryutaroh@newlaptop:~$ systemd-run --unit=myshell --user --scope -p "Delegate=pids memory" /bin/bash
Running scope as unit: myshell.scope

ryutaroh@newlaptop:~$ cat $MY_CGROUP/cgroup.controllers
memory pids

ryutaroh@newlaptop:~$ cat $MY_CGROUP/cgroup.subtree_control

ryutaroh@newlaptop:~$ tail .local/share/lxc/unpriv/config
lxc.net.0.hwaddr = 00:16:3e:e4:ff:b7

lxc.cgroup.devices.deny =
lxc.cgroup.devices.allow =
lxc.cgroup2.devices.deny =
lxc.cgroup2.devices.allow =

# If /sbin/init in guest is systemd, it requires LXC to prepare /sys/fs/cgroup
lxc.mount.auto = cgroup:rw:force
lxc.init.cmd = /bin/bash

ryutaroh@newlaptop:~$ echo +pids > $MY_CGROUP/cgroup.subtree_control
ryutaroh@newlaptop:~$ cat $MY_CGROUP/cgroup.subtree_control
pids

ryutaroh@newlaptop:~$ lxc-start -F -n unpriv
lxc-start: unpriv: cgroups/cgroup2_devices.c: bpf_devices_cgroup_supported: 514 Invalid argument - The bpf device cgroup requires real root
lxc-start: unpriv: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv"
lxc-start: unpriv: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv-1"
lxc-start: unpriv: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv-2"
lxc-start: unpriv: cgroups/cgfsng.c: cg_unified_create_cgroup: 1260 Operation not supported - Could not enable "+pids" controllers in the unified cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor"
lxc-start: unpriv: cgroups/cgfsng.c: cgfsng_monitor_create: 1407 Failed to create cgroup "/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/myshell.scope/lxc.monitor/unpriv-3"
```
The error messages continue 997 times.

**Comment by emojifreak** [comment]
This issue was fixed in `3.2.1+master~20191206-2142-0ubuntu1~eoan` from https://launchpad.net/~ubuntu-lxc/+archive/ubuntu/daily . See https://github.com/lxc/lxc/issues/3198#issuecomment-562808236 for detail.
